admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/web/controller/entity/AdminBasicEntityController.java

```diff
@@ -416,10 +416,38 @@ public String saveEntityJson(HttpServletRequest request, HttpServletResponse res
         if (result.hasErrors()) {
             populateJsonValidationErrors(entityForm, result, json);
         }
+        List<String> dirtyList = buildDirtyList(pathVars, request, id);
+        if (!dirtyList.isEmpty()) {
+            json.with("dirty", dirtyList);
+        }
 
         return json.done();
     }
     
+    public List<String> buildDirtyList(Map<String, String> pathVars, HttpServletRequest request, String id) {
+        List<String> dirtyList = new ArrayList<>();
+        String sectionKey = getSectionKey(pathVars);
+        String sectionClassName = getClassNameForSection(sectionKey);
+        List<SectionCrumb> sectionCrumbs = getSectionCrumbs(request, sectionKey, id);
+        PersistencePackageRequest ppr = getSectionPersistencePackageRequest(sectionClassName, sectionCrumbs, pathVars);
+        ClassMetadata cmd = null;
+        Entity entity = null;
+        try {
```

Comment:
I wouldn't put this in a `try/catch` block. Instead, you should mark the method with `throws ServiceException` and make the caller handle it (in this case, `saveEntityJson()` which is already marked as `throws Exception`.